src/core/grpc_service.proto

```diff
@@ -242,25 +242,82 @@ message SharedMemoryControlRequest
     //@@
     string name = 1;
 
-    //@@  .. cpp:var:: string shared_memory_key
     //@@
-    //@@     The name of the shared memory region that holds the input data
-    //@@     (or where the output data should be written).
+    //@@  .. cpp:var:: message SystemSharedMemoryIdentifier
     //@@
-    string shared_memory_key = 2;
+    //@@     The identifier for this system shared memory region.
+    //@@
+    message SystemSharedMemoryIdentifier
+    {
+      //@@  .. cpp:var:: string shared_memory_key
+      //@@
+      //@@     The name of the shared memory region that holds the input data
+      //@@     (or where the output data should be written).
+      //@@
+      string shared_memory_key = 2;
+    }
+
+    //@@
+    //@@  .. cpp:var:: message CUDASharedMemoryIdentifier
+    //@@
+    //@@     The identifier for this system shared memory region.
+    //@@
+    message CUDASharedMemoryIdentifier
+    {
+      //@@  .. cpp:var:: string shared_memory_key
+      //@@
+      //@@     The name of the system shared memory region that holds the
+      //@@     CUDA shared memory handle.
+      //@@
+      string shared_memory_key = 1;
+
+      //@@  .. cpp:var:: uint64 offset
+      //@@
+      //@@     The offset from the start of the CUDA shared memory region.
+      //@@     start = offset, end = offset + size;
+      //@@
+      uint64 offset = 2;
+
+      //@@  .. cpp:var:: uint64 byte_size
+      //@@
+      //@@     Size of the CUDA shared memory block, in bytes.
+      //@@
+      uint64 byte_size = 3;
```

Comment:
Are the 'offset' and 'byte_size' here the same as those in the outer message? It seems to be a bit confusing without further explanation

Reply:
**offset**: The offset of the cudaIPC handle from the start of the shared memory region.
**byte_size**: Size of the cudaIPC handle in the shared memory block, in bytes.

The cudaIPC handle is stored in shared memory and the 'key' we used for System Shared memory instead must be a specific shared memory region that contains this cudaIPC handle in the case of the CUDA shared memory